src/js/profile/indicator.js

```diff
@@ -67,4 +73,29 @@ export class Indicator extends Component {
 
         }
     }
+
+    addIndicatorText(wrapper, title, indicatorData) {
+        let indicatorNode = indicatorClone.cloneNode(true);
+        $(indicatorTitleClass, indicatorNode).text(title);
+        $(chartDescClass, indicatorNode).text(indicatorData.description);
+        const isLink = !isNull(indicatorData.metadata.url) && !isEmptyString(indicatorData.metadata.url);
```

Comment:
Could you edit dataobjects.js to ensure that metadata.url is never has an empty string? If you find an empty string then convert it to null - that way you don't need complicated boolean statement

Reply:
even so, I like that you're storing the result in a variable/const before using it - helps document the code

Reply:
Edited dataobjects.js and moved metadata extraction to a separate method. Also refactored the existing addIndicatorChart method to use this metadata extraction